client/components/googlePayPayments/oneTimePayment/html/src/app.js

```diff
@@ -0,0 +1,213 @@
+async function onPayPalLoaded() {
+  try {
+    const clientToken = await getBrowserSafeClientToken();
+    const sdkInstance = await window.paypal.createInstance({
+      clientToken,
+      components: ["googlepay-payments"],
+      pageType: "checkout",
+    });
+
+    setupGooglePayButton(sdkInstance);
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+function getPayPalOrder(purchaseAmount) {
+  return {
+    intent: "CAPTURE",
+    purchaseUnits: [
+      {
+        amount: {
+          currencyCode: "USD",
+          value: purchaseAmount,
+          breakdown: {
+            itemTotal: {
+              currencyCode: "USD",
+              value: purchaseAmount,
+            },
+          },
+        },
+      },
+    ],
+    paymentSource: {
+      googlePay: {
+        attributes: {
+          verification: {
+            method: "SCA_WHEN_REQUIRED",
+          },
+        },
+      },
+    },
+  };
+}
+
+function getGoogleTransactionInfo(purchaseAmount, countryCode) {
+  const totalAmount = parseFloat(purchaseAmount);
+  const subtotal = (totalAmount * 0.9).toFixed(2);
```

Comment:
TIL that [toFixed()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toFixed) does rounding. Anytime I see math calculations in JavaScript I get worried about rounding issues. But best I can tell, this works correctly. I did some tests to help validate:

<img width="787" height="645" alt="Screenshot 2025-07-21 at 9 24 23 AM" src="https://github.com/user-attachments/assets/db79220c-ae7e-4423-90b4-16783c4793e2" />